tabrepo/benchmark/experiment/experiment_runner_api.py

```diff
@@ -0,0 +1,409 @@
+                for rep in repetitions_mode_args
+            ]
+        else:
+            assert isinstance(repetitions_mode_args, tuple), (
+                "If `repetitions_mode_args` for 'matrix' is not a list, it must be a tuple"
+            )
+            repetitions_mode_args = [
+                _clean_repetitions_mode_args_for_matrix(repetitions_mode_args)
+            ] * len(tasks)
+        return [[(f, r) for f in e[0] for r in e[1]] for e in repetitions_mode_args]
+
+    if repetitions_mode == "individual":
+        assert repetitions_mode_args is not None, (
+            "If `repetitions_mode` is 'individual', `repetitions_mode_args` must be provided"
+        )
+        assert isinstance(repetitions_mode_args, list), (
+            "If `repetitions_mode` is 'individual', `repetitions_mode_args` must be a list"
+        )
+
+        if isinstance(repetitions_mode_args[0], tuple):
+            assert all(
+                isinstance(rep, tuple)
+                and (len(rep) == 2)
+                and all(isinstance(i, int) for i in rep)
+                for rep in repetitions_mode_args
+            ), (
+                "If `repetitions_mode_args` for 'individual' is a list of tuples, all elements must be tuples of integers of (fold_index, repeat_index) pairs"
+            )
+            repetitions_mode_args = [repetitions_mode_args] * len(tasks)
+
+        # At this point, repetitions_mode_args must be list of lists
+        assert len(repetitions_mode_args) == len(tasks), (
+            "If `repetitions_mode_args` for 'individual' is a list, it must have the same length as `tasks`"
+        )
+        assert isinstance(repetitions_mode_args[0], list), (
+            "Elements of `repetitions_mode_args` for 'individual' must be a list"
+        )
+        assert all(isinstance(rep, list) for rep in repetitions_mode_args), (
+            "If `repetitions_mode_args` for 'individual' is a list, all elements must be lists"
+        )
+        assert all(
+            isinstance(rep, tuple) and (len(rep) == 2) and isinstance(i, int)
+            for e in repetitions_mode_args
+            for rep in e
+            for i in rep
+        ), (
+            "If `repetitions_mode_args` for 'individual' is a list of lists, all inner list elements must be tuples of integers of (fold_index, repeat_index) pairs"
+        )
+
+        return repetitions_mode_args
+
+    raise ValueError(f"Unknown `repetitions_mode` str: {repetitions_mode}")
+
+
+def run_experiments_new(
+    *,
+    output_dir: str,
+    model_experiments: list[Experiment],
+    tasks: list[int | UserTask],
+    repetitions_mode: Literal["TabArena-Lite", "matrix", "individual"],
+    repetitions_mode_args: tuple | list | None = None,
+    run_mode: str = "local",
+    cache_mode: Literal["default", "ignore", "only"] = "default",
+    s3_kwargs: dict | None = None,
+    raise_on_failure: bool = True,
+    debug_mode: bool = False,
+) -> list[dict]:
```

Comment:
Currently I use `ExperimentBatchRunner` for running experiments, whereas you directly call `run_experiments`. Should we have `ExperimentBatchRunner` use `run_experiments_new` instead of `run_experiments`? Or should that be a separate PR?